functions.js

```diff
@@ -0,0 +1,106 @@
+const {dbConnection} = require("./db");
+
+const getAllStations = async (req, res) => {
+    const db = await dbConnection
+    try {
+        const stations = await db.query('SELECT DISTINCT `code`, `name` FROM stations Order By `name` ASC')
+        res.status(200).json({message: "Successfully Retrieved Stations.", data: stations})
+    } catch (e) {
+        res.status(500).json({message: "Unexpected Error", data: []})
+    }
+}
+
+const getJourneys = async (req, res) => {
+    try {
+        // Get Query Params & Check they're populated. Throw Bad Request.
+        const {origin: originSelection, destination: destinationSelection} = req.query
+        if (!originSelection || !destinationSelection) {
+            res.status(400).json({message: "Origin and Destination are required."})
+        } else {
+            // Unique Stations on Dropdown requires selection of all instances of Origins and Destinations...
+            const origins = await getStationInstances(originSelection)
+            const destinations = await getStationInstances(destinationSelection)
+
+            // Extract Lines to See if Single Line is Possible
+            const originLines = origins.map(station => station.line)
+            const destinationLines = destinations.map(station => station.line)
+
+            // Create Pairs of Origins and Destinations. Stations Exist across multiple Lines.
+            const pairs = originLines.reduce((accumulator, line, originIndex) => {
+                const destinationIndex = destinationLines.indexOf(line)
+                if (destinationIndex !== -1) {
+                    accumulator.push([originIndex, destinationIndex])
+                }
+                return accumulator
```

Comment:
Nice reduce! You could of course just use `originLines.filter` with `destinationLines.includes(line)` to shorten this a fair bit.